src/state/CalculationLogics.ts

```diff
@@ -276,20 +292,21 @@ export function getNewCyclesHistory(periodDays: string[]) {
 
   let newCycles: Cycle[] = [
     {
+      // If there is only one cycle in the array, length of this cycle will be 28 (default length)
       cycleLength: 28,
       periodLength: 1,
       startDate: periodDays[0],
     },
   ];
+
+  // In this we form a new "cycles" array
   for (let i = 1; i < periodDays.length; i++) {
     const date = startOfDay(new Date(periodDays[i]));
     const prevDate = startOfDay(new Date(periodDays[i - 1]));
     const diffInDays = differenceInDays(date, prevDate);
 
-    if (diffInDays <= 1) {
-      newCycles[0].periodLength++;
-    } else if (diffInDays <= 2) {
-      newCycles[0].periodLength += 2;
+    if (diffInDays <= 2) {
+      newCycles[0].periodLength += diffInDays === 1 ? 1 : 2;
```

Comment:
Unfortunately I don't understand what's going on here

If `diffInDays` here equals to `0`, `-1` or lesser why we adding `2` here?

The other question if `diffInDays` may be `1` or `2` than we you don't needed in this check here. Instead you just can write `newCycles[0].periodLength += diffInDays`

Reply:
You're right, I fixed it.